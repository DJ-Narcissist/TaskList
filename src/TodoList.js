import React, { useState } from "react";
import Todo from "./Todo";
import NewTodoForm from "./NewTodoForm";

const TodoList = () => {
    const [items, setItems] = useState([]);
    

    const addItem = (newItem) => {
            setItems([...items, newItem]);
    };

    const update = (id, updatedTask) => {
        setItems(items.map((item) => item.id === id ? {...item, task: updatedTask} : item));
    };
    



    const handleDelete = (id) => {
        const updatedItems = items.filter((item) => item.id !== id);
        setItems(updatedItems);
    };

    const itemComponents = items.map(item => (
        <Todo 
        key={items.id}
        id ={item.id}
        task={item.task}
        item={item}
        update={update}
        remove={handleDelete}
        />
    ))
    
    
    return (
        <div>
            <NewTodoForm addItem={addItem} />
            {itemComponents}
        </div>    
    );
};
export default TodoList;