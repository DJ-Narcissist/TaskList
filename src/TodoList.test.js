import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

const addItem = (TodoList, task ="write tests") => {
    const taskInput = TodoList.getByLabelText("Task:");
    fireEvent.change(taskInput, { target: { value: task}});
    const submitButton = TodoList.getByText(" Add Task");
    fireEvent.click(submitButton);
}

it("renders without crashing", () => {
    render (<TodoList />);
});

it ("matches the snapshot", () => {
    const {  asFragment } = render (<TodoList />);
    expect(asFragment()).toMatchSnapshot();
});


it ("can add a task", () => {
    const list = render (<TodoList />);
    addItem(list);

    //expect form to clear and task should appear 
    expect(list.getByLabelText("Task:")).toHaveValue("");
    expect(list.getByText("write tests")).toBeInDocument();
    expect(list.getByText("Edit:")).toBeInDocument();
    expect(list.getByText("X")).toBeInDocument();
});

it ("can edit a task", () => {
    const list = render (<TodoList />);
    addItem(list);

    fireEvent.click(list.getByText("Edit"));
    const editInput = list.getAllByDisplayValue("write tests");
    fireEvent.change(editInput, {target: { value:"sleep" }});
    fireEvent.click(list.getByText("Update!"));

    //expect task show when edited
    expect(list.getByText("sleep")).toBeInDocument();
    expect(list.queryByText("write tests")).not.toBeInDocument();
});

it ("can delete a task", () => {
    const list = render (<TodoList />);
    addItem(list);

    fireEvent.click(list.getByText("X"));

    // Expect task to be deleted
    expect(list.queryByText("write tests")).not.toBeInDocument();
});



