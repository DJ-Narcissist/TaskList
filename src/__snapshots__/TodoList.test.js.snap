// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="key"
      >
        Task:
        <input
          name="task"
          type="text"
          value=""
        />
      </label>
      <button
        type="submit"
      >
        Add Task
      </button>
    </form>
  </div>
</DocumentFragment>
`;
